zerver/lib/user_groups.py

```diff
@@ -669,6 +669,17 @@ def get_direct_memberships_of_users(user_group: UserGroup, members: list[UserPro
 # https://code.djangoproject.com/ticket/28919
 
 
+def get_recursive_subgroups_union_for_groups(user_group_ids: list[int]) -> QuerySet[UserGroup]:
+    cte = With.recursive(
+        lambda cte: UserGroup.objects.filter(id__in=user_group_ids)
+        .values(group_id=F("id"))
+        .union(
+            cte.join(NamedUserGroup, direct_supergroups=cte.col.group_id).values(group_id=F("id"))
+        )
+    )
+    return cte.join(UserGroup, id=cte.col.group_id).with_cte(cte)
+
+
 def get_recursive_subgroups(user_group_id: int) -> QuerySet[UserGroup]:
```

Comment:
This can probably be now be just `get_recursive_subgroups_union_for_groups([user_group_id])` to avoid duplication.